rust/otap-dataflow/crates/engine/src/runtime_pipeline.rs

```diff
@@ -179,15 +221,37 @@ impl<PData: 'static + Debug> RuntimePipeline<PData> {
         node_id: NodeId,
         ctrl_msg: NodeControlMsg,
     ) -> Result<(), Error<PData>> {
-        if let Some(node) = self.get_node(&node_id) {
-            node.send_control_msg(ctrl_msg)
-                .await
-                .map_err(|e| Error::NodeControlMsgSendError {
-                    node: node_id.clone(),
-                    error: e,
-                })
-        } else {
-            Err(Error::UnknownNode { node_id })
+        match self.nodes.get(node_id.index) {
+            Some(ndef) => match ndef.ntype {
+                NodeType::Receiver => {
+                    self.receivers
+                        .get(ndef.inner.index)
+                        .expect("precomputed")
+                        .send_control_msg(ctrl_msg)
+                        .await
+                }
+                NodeType::Processor => {
+                    self.processors
+                        .get(ndef.inner.index)
+                        .expect("precomputed")
+                        .send_control_msg(ctrl_msg)
+                        .await
+                }
+                NodeType::Exporter => {
+                    self.exporters
+                        .get(ndef.inner.index)
+                        .expect("precomputed")
+                        .send_control_msg(ctrl_msg)
```

Comment:
These callsites, I was being asked to disambiguate which call to `send_control_msg` was desired, and I think this skips the `&dyn Node` variation?